lib/hotel.rb

```diff
@@ -0,0 +1,135 @@
+require_relative 'reservation'
+require_relative 'block'
+
+class Hotel
+  attr_reader :reservations, :rooms, :blocks
+  
+  def initialize(reservations: nil, blocks:nil) 
+    @reservations = reservations || []
+    @blocks = blocks || []
+    @rooms = [*1..20]
+  end
+  
+  
+  def make_reservation(start_date:, end_date:, hold_block:false, block_id:nil)
```

Comment:
There's enough going on here that it'd be cleaner to break it up into smaller methods that this method calls.

Reply:
Oh! I see you called that out in your refactors list! Awesome!